### PR TITLE
Add getCodeDetails API to aid partners in migrating off direct quorum access

### PR DESCRIPTION
**Pull request by ChumpChief**
The Fluid app is currently using the code details for more than just code loading, by also adding a "config" into it.  After the code has loaded, during either `ContanerRuntime` instantiation (on container context) or during root component canvas component assignment (on the data store runtime) this config is consulted to alter the runtime behavior of the container code.

Generally speaking this is an antipattern.  The code details shouldn't be used to modify runtime behavior of a container code, as this couples the container code against the code loader and host (it requires the code details format to agree between all three -- normally the agreement is only between the code loader and host).

We're working out alternatives that avoid this pattern, but in the meantime this API will provide a direct migration path for those usages that unblocks our ability to close off direct access to the `Quorum` proposals.

### CODE REVIEW

File: packages/runtime/runtime-definitions/src/dataStoreContext.ts [hunk 1]
```diff
@@ -135,6 +136,15 @@ export interface IContainerRuntimeBase extends
...
     getQuorum(): IQuorum;
 
+    /**
+     * @deprecated This method is provided as a migration tool for customers currently reading the code details
+     * from within the Container by directly accessing the Quorum proposals.  The code details should not be accessed
+     * from within the Container as this requires coupling between the container contents and the code loader.
+     * Direct access to Quorum proposals will be removed in an upcoming release, and in a further future release this
+     * migration tool will be removed.
+     */
+    getSpecifiedCodeDetails?(): IFluidCodeDetails | undefined;
```
Comment (anthony-murphy) [inline packages/runtime/runtime-definitions/src/dataStoreContext.ts:146]: i'm looking here

Comment (anthony-murphy) [reply packages/runtime/runtime-definitions/src/dataStoreContext.ts:146]: so this will be removed?

**Review by anthony-murphy — APPROVED** [review]

File: packages/runtime/runtime-definitions/src/dataStoreContext.ts (hunk 1 above)
Comment (ChumpChief) [reply packages/runtime/runtime-definitions/src/dataStoreContext.ts:146]: For context, we were discussing that the runtime should really never have access to the specified code details, which I agree with.  I think we probably can sufficiently cover this scenario without adding this API here by ensuring the config is "stamped" on the container in some way (e.g. create the correct type of component in current flow, store as data in the generic root component in future flow).  I'm investigating this further to validate.

Comment (ChumpChief) [reply packages/runtime/runtime-definitions/src/dataStoreContext.ts]: Removed in latest.

**Pull request merged**